packages/rum/src/domain/record/record.ts

```diff
@@ -74,34 +58,18 @@ export function record(options: RecordOptions): RecordAPI {
 
   const { stop: stopObservers, flush: flushMutationsFromObservers } = initObservers(configuration, {
     lifeCycle: options.lifeCycle,
-    configuration,
     elementsScrollPositions,
-    inputCb,
-    mediaInteractionCb: (p) =>
-      emitAndComputeStats(assembleIncrementalSnapshot<MediaInteractionData>(IncrementalSource.MediaInteraction, p)),
-    mouseInteractionCb: (mouseInteractionRecord) => emitAndComputeStats(mouseInteractionRecord),
-    mousemoveCb: (positions, source) =>
-      emitAndComputeStats(assembleIncrementalSnapshot<MousemoveData>(source, { positions })),
-    mutationCb,
-    scrollCb: (p) => emitAndComputeStats(assembleIncrementalSnapshot<ScrollData>(IncrementalSource.Scroll, p)),
-    styleSheetCb: (r) =>
-      emitAndComputeStats(assembleIncrementalSnapshot<StyleSheetRuleData>(IncrementalSource.StyleSheetRule, r)),
-    viewportResizeCb: (d) =>
-      emitAndComputeStats(assembleIncrementalSnapshot<ViewportResizeData>(IncrementalSource.ViewportResize, d)),
-    frustrationCb: (frustrationRecord) => emitAndComputeStats(frustrationRecord),
-    focusCb: (data) =>
-      emitAndComputeStats({
-        data,
-        type: RecordType.Focus,
-        timestamp: timeStampNow(),
-      }),
-    visualViewportResizeCb: (data) => {
-      emitAndComputeStats({
-        data,
-        type: RecordType.VisualViewport,
-        timestamp: timeStampNow(),
-      })
-    },
+    inputCb: emitAndComputeStats,
+    mediaInteractionCb: emitAndComputeStats,
+    mouseInteractionCb: emitAndComputeStats,
+    mousemoveCb: emitAndComputeStats,
+    mutationCb: emitAndComputeStats,
+    scrollCb: emitAndComputeStats,
+    styleSheetCb: emitAndComputeStats,
+    viewportResizeCb: emitAndComputeStats,
+    frustrationCb: emitAndComputeStats,
+    focusCb: emitAndComputeStats,
+    visualViewportResizeCb: emitAndComputeStats,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ if they're all the same, can't we just have one generic `cb` property?

Reply:
Good point! I'm even wondering if we still need the observer.ts. I'll make some changes and let you know :)

Reply:
So:
- I moved the [observers initialization up in record()](https://github.com/DataDog/browser-sdk/blob/02065577bb1aeae6ccfffb738dd3132c2f3e16a8/packages/rum/src/domain/record/record.ts#L75) and removed the initObservers() wrapper. 
- Harmonize observers to return a `stop` function 
Let me know what you think